napari/utils/events/containers/_dict.py

```diff
@@ -51,6 +52,15 @@ def __iter__(self) -> Iterator[_T]:
     def __repr__(self):
         return str(self._dict)
 
+    if sys.version_info < (3, 8):
```

Comment:
Thanks for working on this @lukasz-migas !

Even if code inside a class is executed and this works, I find it more readable to do something like:

```python
        # We've explicitly added __hash__ for python < 3.8 because otherwise
        # nested evented dictionaries fail tests.
        # This can be removed once we drop support for python < 3.8
        # see: https://github.com/napari/napari/pull/2994#issuecomment-877105434
        def __hash__(self):
            if sys.version_info < (3, 8):
                return hash(frozenset(self))

            return super().hash()
```

Reply:
Slightly more readable slightly less performant, yeah?

I proposed the class level check.
not married to it, but the rationale was that hashing is pretty critical and no need to check every time

Reply:
Happy to make this change but the `if` statement will have to be evaluated each time a hash is requested. I doubt this will have much of an impact but still.

Reply:
@goanpeca do you have strong feelings here? I tend to agree with "slightly more readable, slightly less performant" - not sure which one we should weight more heavily - personally I really don't mind.

Reply:
No strong feelings, feel free to merge.